### Useful arithmetic

### Gauss' Formula

$ \sum_{i=0}^{n}i,\sum_{i=1}^{n}i=\frac{n(n+1)}{2} $

If you were asked to add all the numbers from 1-100, you might make a sequence in ascending order:

$ s=\{1,2,3,...,99,100\} $

If you were to reverse the set:

$ s'=\{100,99,98,...,2,1\} $

And then add every element by order of place in the sequence:

$ 2s=\{101,101,101,...,101,101\} $

$ 2s=100(101) $

$ s=\frac{100(101)}{2} $

The generalisation is:

$ \frac{n(n+1)}{2} $

### Difference of squares

Consider $ a^2 - b^2 $

Then this problem can be rewritten as:

$ (a - b) = \frac{(a^2 - b^2)}{(a - b)} $

It's a simple trick but worth remembering.

### Telescoping sums

$ \sum_{i=2}^{n=max(x)}x-x_{i-1}=-x_{1}+x_{n} $

If you need to find any sum where each sum step is a subtraction, like:

$ \sum_{i}^{n}i-i_{i-1} $

Note that this sum is not the only example.

Then a concrete example is:

$ x=\{1,2,3,4,5\} $

$ \sum_{i=2}^{n=5}x_{i}-x_{i-1} $

And expanded this is:

$ (2-1)+(3-2)+(4-3)+(5-4) $

$ 2-1+3-2+4-3+5-4 $

If we remove like terms:

$ -1+5 $

### Any infinitely repeating decimal is a rational number

Just a concrete example will suffice:

Given

$ x=1.2(9) $

Then

$ \begin{array}{rrrc}
10x & = & 12.9(9) & \text{ }\\
x & = & 1.2(9) & -\\
9x & = & 11.7(0) & =
\end{array} $

$ \therefore x=\frac{11.7}{9}=1.3 $

### Completing the square

Given a quadratic function in the form $ ax^{2}+bx+c $

We can find $ x $ by completing the square.

A square is in the form: $ (x+y)^{2} $, where $ y $ is known

These are squares:

$ x^{2}+4x+4 = (x+2)(x+2) = (x+2)^{2} $

$ x^{2}+6x+9 = (x+3)(x+3) = (x+3)^{2} $

$ x^{2}+8x+16 = (x+4)(x+4) = (x+4)^{2} $

For an example, this is not a square: 

$ f(x)=3x^{2}+18x+26 $

To complete the square of $ f(x) $, first, put the constant on the right and then factor out the multiple of $ x^2 $:

$ 3(x^{2}+6x)=-26 $

Next, expand the square with $ z $ unknown: 

$ f(x) = x^{2}+6x+z = (x+3)(x+3) = (x+3)^{2} $

$ z = \left( \frac{6}{2} \right)^2 = 9 $

So you add the 9 to the left side, but also add the same amount $ 9*3 = 27 $ to the right side.

$ 3(x^{2}+6x+9) = 1 $

$ 3(x+3)^{2}-1 = 0 $

We now have a function with a single x that we can find the value of:

$ 3(x+3)^{2}=1 $

$ (x+3)^{2}=\frac{1}{3} $

$ (x+3)=\pm\sqrt{\frac{1}{3}} $

$ x=\pm\sqrt{\frac{1}{3}}-3 $

### The square of a negative

This is extremely basic, but it has caused issues for me enough times that I feel I must write it down.

$ -x^{2}\sim(-x)^{2} $

$ -x^2 = -(x)^2 $

Use parentheses in substitutions to avoid this.

### Partial fraction decomposition

If you have a fraction of two polynomials, we can decompose it to two individual parts. This is conceptually the opposite of the product of two fractions.

This is very useful for integration, since integrating a quotient of complex parts can be hard. Integrating two quotients of relatively simple parts is much simpler.

The requirement is for the numerator and denominator to be polynomials. I think the denominator needs to be a polynomial of degree 2 or more.

When there are repeated terms in the denominator, increase their degree.

eg:

$ \frac{3x+5}{(x+1)(x-3)^2} = \frac{A}{(x+1)} + \frac{B}{(x-3)} + \frac{C}{(x-3)^2}$

You must always make sure that the degree of the numerator is one less than the denominator.

eg:

$ \frac{2x^2 -x +4}{x^3 + 4x} = \frac{Ax + B}{x^2 + 4} + \frac{C}{x} $

Here's an end-to-end example:

$ \frac{x + 8}{x^2 - 6x -7} $

$ \frac{x + 8}{(x + 7)(x - 1)} = \frac{A(x - 1)}{(x + 7)(x - 1)} + \frac{B(x + 7)}{(x + 7)(x - 1)} = \frac{A}{(x + 7)} + \frac{B}{(x - 1)} $

Above, the term on the right is our desired goal to find.

You can eliminate the quotients by multiplying the denominator from the LHS to all denominators.

$ (x + 8) = A(x - 1) + B(x + 7) $

You can solve those as simultaneous equations or as a system of linear equations.

$ (x + 8) = Ax - A + Bx + B7 $

To use row reductions, we would need to assign the columns as the unknown variables $ A,B $. Treat x as a constant =1.

$ \begin{matrix} A & B & = \\ 1 & 1 & 1 \\ -1 & 7 & 8 \end{matrix} $

In [1]:
M = [1 1 1;-1 7 8]

M =

   1   1   1
  -1   7   8



In [2]:
M(2,:) = M(2,:) + M(1,:);

In [3]:
M(2,:) = M(2,:)/8;

In [4]:
M(1,:) = M(1,:) - M(2,:)

M =

   1.00000   0.00000  -0.12500
   0.00000   1.00000   1.12500



Then this is the partial fraction decomposition:

$ \frac{-0.125}{(x + 7)} + \frac{1.125}{(x - 1)} $

### Recurring square roots

Say you have a problem that looks like:

$ \sqrt{n+\sqrt{n+\sqrt{n+...}}}$

$ n \in R $ (not sure if it works with complex numbers).

Then:

$ \sqrt{n+\sqrt{n+\sqrt{n+...}}} = x $

$ n+\sqrt{n+\sqrt{n+\sqrt{n + ...}}} = x^2 $

$ n + x = x^2 $

$ x^2 -x -n = 0 $

Then solve the quadratic equation.